neo.UnitTests/UT_Helper.cs

```diff
@@ -38,5 +43,271 @@ public void ToScriptHash()
+                Weight = w
+            });
+            var sum = BigInteger.Zero;
+            foreach (Result res in ret)
+            {
+                sum = BigInteger.Add(res.Weight, sum);
+            }
+            sum.Should().Be(BigInteger.Zero);
+
+            var w3 = new Woo
+            {
+                Value = 3
+            };
+
+            var list2 = new List<Woo>
+            {
+                w1, w2, w3
+            };
+            var ret2 = list2.WeightedFilter(0.3, 0.4, p => p.Value, (p, w) => new Result
+            {
+                Info = p,
+                Weight = w
+            });
+            sum = BigInteger.Zero;
+            foreach (Result res in ret2)
+            {
+                sum = BigInteger.Add(res.Weight, sum);
+            }
+            sum.Should().Be(BigInteger.Zero);
+
+            Action action1 = () => list2.WeightedFilter(-1, 0.4, p => p.Value, (p, w) => new Result
+            {
+                Info = p,
+                Weight = w
+            }).WeightedAverage(p => p.Weight, p => p.Weight);
+            action1.Should().Throw<ArgumentOutOfRangeException>();
+
+            Action action2 = () => list2.WeightedFilter(0.2, 1.4, p => p.Value, (p, w) => new Result
+            {
+                Info = p,
+                Weight = w
+            }).WeightedAverage(p => p.Weight, p => p.Weight);
+            action2.Should().Throw<ArgumentOutOfRangeException>();
+
+            Action action3 = () => list2.WeightedFilter(0.8, 0.3, p => p.Value, (p, w) => new Result
+            {
+                Info = p,
+                Weight = w
+            }).WeightedAverage(p => p.Weight, p => p.Weight);
+            action3.Should().Throw<ArgumentOutOfRangeException>();
+
+            Action action4 = () => list2.WeightedFilter(0.3, 0.8, p => p.Value, (p, w) => new Result
+            {
+                Info = p,
+                Weight = w
+            }).WeightedAverage(p => p.Weight, p => p.Weight);
+            action4.Should().Throw<ArgumentOutOfRangeException>();
+
+            Action action5 = () => list2.WeightedFilter(0.3, 0.6, null, (p, w) => new Result
+            {
+                Info = p,
+                Weight = w
+            }).WeightedAverage(p => p.Weight, p => p.Weight);
+            action5.Should().Throw<ArgumentNullException>();
+
+            list2.WeightedFilter(0.3, 0.3, p => p.Value, (p, w) => new Result
+            {
+                Info = p,
+                Weight = w
+            }).WeightedAverage(p => p.Weight, p => p.Weight).Should().Be(0);
+
+
+            var list3 = new List<Woo>();
+            list3.WeightedFilter(0.3, 0.6, p => p.Value, (p, w) => new Result
+            {
+                Info = p,
+                Weight = w
+            }).WeightedAverage(p => p.Weight, p => p.Weight).Should().Be(0);
+
+            list2 = null;
+            Action action6 = () => list2.WeightedFilter(0.3, 0.4, p => p.Value, (p, w) => new Result
```

Comment:
Sounds good. Could these actions checks be generalized into a function for becoming a little bit more compact?